### ISSUE TITLE
Template and file changes are delayed

### ISSUE THREAD

**Issue by jasonmccallister** [craftcms/nitro#168]
There are some reports of getting random slow load times (sometimes up to 20 seconds) for pages (especially noticeable with Twig templates). 

Underneath the hood, `multipass` uses `sshfs` to "mount" the directories into the virtual machine. There is a default 20 second cache timeout set when using `sshfs` through `multipass`. Right now there is no way to override the settings passed to `sshfs` when using `multipass mount`. However, there is a ticket to add support for disabling the `sshfs` cache here: https://github.com/canonical/multipass/issues/1502.

As an experiment, we used the script provided by @RoelTiny (thank you) to create a NFS mount.

Here are the steps to reproduce the machine with the NFS mount:

1. Create a new machine with `nitro --machine testing-nfs init`
2. Once that machine has been created, do not mount any directories using `nitro add`
3. `nitro ssh` into the machine and run `sudo apt update && sudo apt install -y nfs-common`
3. Run the scripts here (https://github.com/craftcms/nitro/blob/feature/mount-performance/scripts/mount.sh ) passing in the workspace to mount as the first argument and the directory to create: `cd scripts && ./mount.sh /Users/jasonmccallister/dev testing-nfs-mounts`
4. Create the nginx site manually `sudo cp /opt/nitro/nginx/template.conf /etc/nginx/sites-available/craft.nitro` edit the contents for the webroot and php version,
5. Symlink the new site `sudo ln -s /etc/nginx/sites-available/craft.nitro /etc/nginx/sites-enabled/`
6. Reload nginx `sudo nginx reload`
7. Edit the hosts file to point the new domain to the VM ip e.g. `192.168.64.76    craft.nitro`

```
# /etc/exports on host machine
/Users/jasonmccallister/dev -mapall=jasonmccallister 192.168.64.76
```

**Comment by jasonmccallister** [comment]
Some additional research: https://blog.ja-ke.tech/2019/08/27/nas-performance-sshfs-nfs-smb.html

**Comment by rostockahoi** [comment]
Thank you for the research @jasonmccallister. Does the file sync to the mount "feel" instant after applying this workaround?

**Comment by jasonmccallister** [comment]
@rostockahoi it is instant and changes show up quickly. However, NFS mounting does not seem well supported on Windows and if we stop using the mount command, we are going to have more support and issues than it might be worth. The alternative, that we are hacking on now, is modifying the amount of time that multipass mount caches sshfs mounts. 

Performance with NFS looks promising, but I'm not sure on the support required but if we can't get options/sshfs to be performant, we are probably going to have to switch to NFS.

**Comment by Coysh** [comment]
Any update on this? Or are we still waiting for a multipass update?

**Comment by angrybrad** [comment]
@Coysh you can follow along on the PR here: https://github.com/canonical/multipass/pull/1605

**Comment by jasonmccallister** [comment]
There is a macOS and Linux build [here](https://github.com/canonical/multipass/pull/1654#issuecomment-662477956) that lowers the caching to 3 seconds. This is quite an improvement from my setup and wanted to pass this on for testing before the multipass team makes it official.

If either of you have the time to try this out and provide feedback here it would be helpful! Thank you!

**Comment by Coysh** [comment]
@jasonmccallister just tried out the latest build and seems to be significant improvement for me. Thanks!

**Comment by nordentwickler** [comment]
I can confirm, it is faster and working with browsersync and webpack feels snappier.

**Comment by jasonmccallister** [comment]
Thank you @Coysh and @nordentwickler for the feedback!

**Comment by jasonmccallister** [comment]
This has been resolved with Multipass 1.4.0 and we just tagged 1.0.0! Closing this out!
